crates/router/src/core/utils.rs

```diff
@@ -992,16 +997,19 @@ pub async fn get_profile_id_from_business_details(
     merchant_account: &domain::MerchantAccount,
     request_profile_id: Option<&String>,
     db: &dyn StorageInterface,
+    should_validate: bool,
```

Comment:
validate only in case of payments and mca create, in other cases validation is not needed

Reply:
Is it required in list payment methods also?

Reply:
No its not required in payment methods list. The entry point is payments create and once a profile id is given for a payment, it cannot be changed in any other flow